### DISCUSSION TITLE
Delete a block and "undo" cannot bring it back

### DISCUSSION THREAD

**Discussion by yifanyin** [athensresearch/athens#951] [Archive]
Has anyone encountered this bug/feature? Once I select a whole block (nodes included, with a blue shade over the block) and delete it. "Undo" will not bring back the block. The same goes with cutting the whole block (see my question [here](https://discord.com/channels/708122962422792194/802652338619285594/830744523268292659)). It is a bit inconvenient if the deleted block has several references attached to it.

**Comment by jefftangx** [comment]
Hi @yifanyin just repro'd that Undo doesn't work when cutting or deleting a whole block. @arkRedM worked on this mostly, but definitely a bug!

**Comment by solarkraft** [comment]
I'd put a little bounty on this, it's really annoying :(

**Reply by jefftangx** [reply]
@arkRedM do you think it'd be difficult to address this?

@jsmorabito can you make a bounty for this when you have a moment?

**Reply by jsmorabito** [reply]
https://gitcoin.co/issue/athensresearch/athens/987/100025562 lmk if i should modify any part of the bounty post

**Reply by yifanyin** [reply]
Would it be possible to tie discussion #952 with it? Feel like it's somewhat connected.
